include/alpaka/core/UniqueId.hpp

```diff
@@ -108,7 +108,7 @@ namespace alpaka
             auto constexpr reader(
                 float,
                 flag<N>,
-                std::size_t R = reader(std::size_t{0u}, flag<N-1u>{}))
+                std::size_t R = reader(std::size_t(0), flag<N-1u>()))
```

Comment:
I think I changed it because it not compile with `{}` but **I will check it again**.